blog/2022-02-03-user-avatar.mdx

```diff
@@ -0,0 +1,299 @@
+---
+slug: manage-user-avatar
+
+title: All you need to know to manage user avatar 👤 in React
+
+authors: [jcizmar]
+
+image: "/img/blog/2022-01-20-super-fast-react-localizaton-with-i18next-and-tolgee/cover.jpg"
+
+tags: [react, i18n, i18next, javascript, typescript, integration]
+---
+
+Most of the information systems using user authentication also work with some kind of avatars - images enabling other
+users to see who commented, posted or made anything else in the system.
+When we want to be able to show user avatars, we need to enable users to upload their avatar or generate some default.
+In this article I am going to show you how to:
+- generate user avatar by their unique ID using `jdenticon`
+- select avatar from file
+- crop selected avatar using `cropper.js` and `react-cropper` libraries
+
+## Generating default avatar
+When user hasn't selected their own avatar, we have few options what to show instead of it.
+We can show nothing, use some placeholder image 👤 or **generate** some pretty cool image. 😎
+
+There are multiple avatar generators in npm. I chose [`jdenticon`](https://www.npmjs.com/package/jdenticon), because I like the images it produces.
+
+![Identicon Example 1](/img/blog/avatar/jdenticon.png)
+
+You can also find this [nice icon designer](https://jdenticon.com/icon-designer.html) in their docs, so you can
+customize the output. But I am going to stick with defaults.
+
+So lets see how to use the library to generate the image.
+```tsx
+import { ComponentProps, FC, useState } from 'react';
+
+export const AutoAvatar: FC<
+  ComponentProps<'img'> & { userId: number; size: number }
+> = ({ userId, size, ...imgProps }) => {
+  const [base64, setBase64] = useState(undefined as string | undefined);
+
+  // using dynamic import to save some loading
+  import('jdenticon').then(({ toSvg }) => {
+    const svgString = toSvg(userId, size);
+    const base64 = Buffer.from(svgString).toString('base64');
+    setBase64(base64);
+  });
+
+  return base64 ? (
+    <div style={{ backgroundColor: 'rgb(225,225,225)', display: 'flex' }}>
+      <img
+        {...imgProps}
+        src={`data:image/svg+xml;base64,${base64}`}
+        alt={'User Avatar'}
+      />
+    </div>
+  ) : (
+    <div style={{ width: size, height: size, display: 'inline-block' }}>
+      Loading...
+    </div>
+  );
+};
+```
+What I am doing here is loading the `jdenticon` library using dynamic import. The library has ca. 45 kb, so it is good
```

Comment:
"ca." looks weird to me, I think "approx." would be better
https://forum.thefreedictionary.com/postst35264_Approximately-or-Circa.aspx